src/components/RevokeContainer/RevokeContainer.tsx

```diff
@@ -0,0 +1,110 @@
+  const [revokeStep, setRevokeStep] = useState(1);
+  const [documentUploadState, setDocumentUploadState] = useState(DocumentUploadState.INITIALIZED);
+
+  useEffect(() => {
+    if (revokeDocuments.length > 0) {
+      const isDocumentRevokable = utils.isDocumentRevokable(revokeDocuments[0]);
+
+      const validateDocument = async () => {
+        const network = config?.network || "homestead";
+        let isDocumentValid = true;
+
+        if (network !== "local") {
+          const verify = verificationBuilder(openAttestationVerifiers, { network: network });
+          const fragments = await verify(revokeDocuments[0]);
+          isDocumentValid = isValid(fragments);
+        }
+
+        if (!isDocumentValid) {
+          setErrorMessage("Error: Document is not valid, please use a valid document.");
+          setRevokeDocuments([]);
+          setDocumentUploadState(DocumentUploadState.ERROR);
+        } else {
+          setErrorMessage("");
+          setRevokeStep(2);
+          setDocumentUploadState(DocumentUploadState.DONE);
+        }
+      };
+
+      if (!isDocumentRevokable) {
+        setErrorMessage("Error: Document is not revokable, please use a revokable document.");
+        setRevokeDocuments([]);
+        setDocumentUploadState(DocumentUploadState.ERROR);
+      } else {
+        validateDocument();
+      }
+    }
+  }, [config?.network, revokeDocuments]);
+
+  if (!config) {
+    return <Redirect to="/" />;
+  }
+
+  const revokingDocument = () => {
+    setRevokeStep(3);
+    setShowConfirmationModal(false);
+  };
+  const revokeAnotherDocument = () => {
+    setRevokeStep(1);
+    setRevokeDocuments([]);
+    setFileName("");
+    setErrorMessage("");
+    setShowConfirmationModal(false);
+    setDocumentUploadState(DocumentUploadState.INITIALIZED);
+  };
+
+  return (
+    <>
+      <RevokeConfirmationModal
+        revokingDocument={revokingDocument}
+        show={showConfirmationModal}
+        closeRevokeConfirmationModal={() => setShowConfirmationModal(false)}
+      />
+      {revokeStep === 1 && (
+        <RevokeDocumentDropZone
+          setRevokeDocuments={setRevokeDocuments}
+          errorMessage={errorMessage}
+          setFileName={setFileName}
+          documentUploadState={documentUploadState}
+          setDocumentUploadState={setDocumentUploadState}
+        />
+      )}
+      {revokeStep === 2 && (
+        <RevokeDocumentTileArea
+          revokeDocuments={revokeDocuments}
+          fileName={fileName}
+          onShowConfirmation={() => setShowConfirmationModal(true)}
+          documentUploadState={documentUploadState}
+          onBack={revokeAnotherDocument}
+        />
+      )}
+      {revokeStep === 3 && (
+        <ProcessDocumentScreen
+          config={config}
+          revokeDocuments={revokeDocuments}
+          processAnotherDocument={revokeAnotherDocument}
+          fileName={fileName}
+          type={QueueType.REVOKE}
+        />
+      )}
+    </>
+  );
```

Comment:
we are following such a pattern for publishing as well right?
rendering a component based on the step of the flow?

Reply:
hmm.. not exactly, we are giving a route to each step for publishing

for now this way is okay, although we should really look up and create a wizard component for ourselves

Reply:
erm, nope 
step 1 ->  we use path: /forms-selection
step 2 -> we use path: /form
step 3 -> we use path: /publish

but for form we look out for if the form data is there if not we redirect it to login page.
Would you want me to update the logic for this part to have multiple paths?
should we

Reply:
no for now its okay, we'll revisit in the future